File: src/types/automation.ts
export type WatchStatus = "active" | "paused" | "disabled";
export type AutomationRunStatus = "success" | "error";
export type ScheduleCadence = "hourly" | "weekly";
export type ScheduleStatus = "active" | "paused" | "disabled";

export interface AutomationWatch {
  id: string;
  workflow_id: string;
  watch_path: string;
  recursive: boolean;
  file_glob: string;
  status: WatchStatus;
  debounce_ms: number;
  stability_ms: number;
  created_at: string;
  updated_at: string;
}

export interface AutomationRun {
  id: string;
  watch_id: string;
  workflow_id: string;
  trigger_file_path: string;
  trigger_event_id: string;
  status: AutomationRunStatus;
  started_at: string;
  ended_at: string;
  duration_ms: number;
  error_message?: string;
}

export interface AutomationSchedule {
  id: string;
  workflow_id: string;
  cadence: ScheduleCadence;
  hourly_interval?: number;
  weekly_days: string[];
  hour: number;
  minute: number;
  status: ScheduleStatus;
  last_run_at?: string;
  next_run_at: string;
  created_at: string;
  updated_at: string;
}

export interface AutomationScheduleRun {
  id: string;
  schedule_id: string;
  workflow_id: string;
  trigger_event_id: string;
  status: AutomationRunStatus;
  started_at: string;
  ended_at: string;
  duration_ms: number;
  error_message?: string;
}

File: src-tauri/src/models/automation.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
pub enum WatchStatus {
    Active,
    Paused,
    Disabled,
}

impl WatchStatus {
    pub fn as_str(&self) -> &'static str {
        match self {
            Self::Active => "active",
            Self::Paused => "paused",
            Self::Disabled => "disabled",
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AutomationWatch {
    pub id: String,
    pub workflow_id: String,
    pub watch_path: String,
    pub recursive: bool,
    pub file_glob: String,
    pub status: WatchStatus,
    pub debounce_ms: i64,
    pub stability_ms: i64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
pub enum RunStatus {
    Success,
    Error,
}

impl RunStatus {
    pub fn as_str(&self) -> &'static str {
        match self {
            Self::Success => "success",
            Self::Error => "error",
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AutomationRun {
    pub id: String,
    pub watch_id: String,
    pub workflow_id: String,
    pub trigger_file_path: String,
    pub trigger_event_id: String,
    pub status: RunStatus,
    pub started_at: DateTime<Utc>,
    pub ended_at: DateTime<Utc>,
    pub duration_ms: i64,
    pub result_summary: Option<String>,
    pub error_message: Option<String>,
}
